packages/core/src/utils/Cursor.ts

```diff
@@ -57,21 +57,23 @@ export class Cursor<Entity extends object, Hint extends string = never, Fields e
 
   readonly hasPrevPage: boolean;
   readonly hasNextPage: boolean;
+  readonly totalCount: number | null;
```

Comment:
better to keep this optional instead of allowing `null` values
```suggestion
  readonly totalCount?: number;
```

and ideally this will be conditionally mapped so the property is not defined only when `includeCount: false`, I can take a look at that myself if you have no idea how to do that with TS

Reply:
I will let you do that, I am not sure how to tackle this :).